networklibrary/src/main/java/com/eis/communication/network/SMSNetSubscribers.java

```diff
@@ -0,0 +1,43 @@
+package com.eis.communication.network;
+
+import com.eis.smslibrary.SMSPeer;
+
+import java.util.ArrayList;
+
+/**
+ * Concrete implementation of a {@link INetSubscribers} interface
+ *
+ * @author Marco Cognolato
+ */
+public class SMSNetSubscribers implements INetSubscribers<SMSPeer>  {
+
+    private ArrayList<SMSPeer> subscribers = new ArrayList<>();
+
+    /**
+     * Adds a subscriber to this network
+     * @param subscriber The subscriber to add to the net
+     * @throws IllegalArgumentException If subscriber is null
+     */
+    public void addSubscriber(SMSPeer subscriber){
```

Comment:
What about a `@NonNull final` then? To help to avoid null subscribers

Reply:
`@NonNull` doesn't force any constraint for null values, so I decided to handle it myself. I'll still add the `@NonNull` to let the user know at compile time though 👍